src/tunacode/tools/bash.py

```
@@ -88,8 +171,42 @@ async def bash(
         await _cleanup_process(process)
 
 
+async def _execute_bash(
+    tool_call_id: str,
+    args: JsonObject,
+    signal: asyncio.Event | None,
+    on_update: AgentToolUpdateCallback,
+) -> AgentToolResult:
+    _ = (tool_call_id, on_update)
+    if signal is not None and signal.is_set():
+        raise UserAbortError("Tool execution aborted: bash")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)